Sources/UserDefaultsGeneratorCore/Generator.swift

```diff
@@ -0,0 +1,81 @@
+//
+//  Generator.swift
+//  CYaml
+//
+//  Created by Yudai Hirose on 2019/09/18.
+//
+
+import Foundation
+
+public protocol Generator {
+    func generate() throws
+}
+
+public struct GeneratorImpl: Generator {
+    let configurations: [Configuration]
+    let outputPath: URL
+    public init(configurations: [Configuration], outputPath: URL) {
+        self.configurations = configurations
+        self.outputPath = outputPath
+    }
+    public func generate() throws {
+        let content = enumDefinition() + "\n" + userDefaultsExtensions()
+        try content.write(to: outputPath, atomically: true, encoding: .utf8)
+    }
+    
+    func enumDefinition() -> String {
+        let caseMap: (Configuration) -> String = { configuration in
+            switch configuration.key {
+            case nil:
+                return "case " + configuration.name
+            case let key?:
+                return "case " + configuration.name + " = " + "\"\(key)\""
+            }
+        }
+        return configurations
+            .grouped { $0.type }
+            .map { grouped in
+                let values = grouped.value.map(caseMap).joined(separator: "\n")
+                return """
+                public enum UDG\(grouped.key.rawValue)Key: String {
+                \(values)
+                }
+                """
+            }
+            .joined(separator: "\n")
+    }
+    
+    func userDefaultsExtensions() -> String {
```

Comment:
```suggestion
    private func userDefaultsExtensions() -> String {
```

Reply:
In next PR, to write unit test for enumDefinition. So, I decide to keep internal access control.